lib/maplibre.ex

```diff
@@ -213,6 +262,26 @@ defmodule MapLibre do
     end
   end
 
+  defp validate_data!(data, coordinates) do
+    if Table.Reader.impl_for(data),
+      do: validate_coordinates(coordinates),
+      else: raise(ArgumentError, "unsupported data")
+  end
+
+  defp validate_coordinates(coordinates) do
+    case coordinates do
+      {format, column} when format in [:lng_lat, :lat_lng] and is_binary(column) ->
+        nil
+
+      {format, [lng, lat | []]}
+      when format in [:lng_lat, :lat_lng] and is_binary(lng) and is_binary(lat) ->
+        nil
+
+      _ ->
+        raise(ArgumentError, "unsupported coordinates format")
```

Comment:
Please include the expected formats in the error message

```suggestion
        raise(ArgumentError, "unsupported coordinates format. Expect a tuple of two elements yada yad yada")
```